### ISSUE TITLE
ArrayIndexOutOfBoundsException when turning off page cache

### ISSUE THREAD

**Issue by bachmanm** [neo4j/neo4j#4333]
Try setting dbms.pagecache.memory=0 or some low number (1k), Neo4j 2.2.0 fails to start in all modes (embedded, server) with

```
java.lang.ArrayIndexOutOfBoundsException: 0
    at org.neo4j.io.pagecache.impl.muninn.MuninnPageCache.evictPages(MuninnPageCache.java:739)
    at org.neo4j.io.pagecache.impl.muninn.MuninnPageCache.continuouslySweepPages(MuninnPageCache.java:674)
    at org.neo4j.io.pagecache.impl.muninn.EvictionTask.run(EvictionTask.java:39)
    at org.neo4j.io.pagecache.impl.muninn.BackgroundTask.run(BackgroundTask.java:45)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

**Comment by cleishm** [comment]
Hey Michal,

Great catch. I'll pass that along to our team.

To be clear though - you can't turn off the page cache. It's an essential part of Neo4j, as it does need to bring data off disk and into memory somewhere :) Likely there should be a minimum size that the configuration allows.

Cheers,
Chris

**Comment by bachmanm** [comment]
Fair enough. This was for perf testing to show differences between reading data from disk, from low-level cache, and high-level cache...

Cheers!

**Comment by chrisvest** [comment]
This will be fixed in #4418
